src/screencast/wlr_screencast.c

```diff
@@ -19,14 +19,27 @@
 #include "xdpw.h"
 #include "logger.h"
 
+static void wlr_frame_buffer_destroy(struct xdpw_screencast_instance *cast) {
+	// This check isn't needed
+	if (cast->simple_frame.data != NULL) {
+		munmap(cast->simple_frame.data, cast->simple_frame.size);
+		cast->simple_frame.data = NULL;
+	}
+	// This one is
+	if (cast->simple_frame.buffer != NULL) {
+		wl_buffer_destroy(cast->simple_frame.buffer);
+		cast->simple_frame.buffer = NULL;
+	}
+}
+
 void xdpw_wlr_frame_free(struct xdpw_screencast_instance *cast) {
 	zwlr_screencopy_frame_v1_destroy(cast->wlr_frame);
 	cast->wlr_frame = NULL;
-	munmap(cast->simple_frame.data, cast->simple_frame.size);
-	cast->simple_frame.data = NULL;
 	// TODO: reuse this buffer unless we quit or error out
```

Comment:
This TODO can be removed, correct?

Reply:
Sure. I would remove it, when this draft changed to a PR.

Reply:
Go ahead and pull this todo. I think we're good to merge.

Reply:
Todo removed.